src/app/modules/group/components/associated-activity/associated-activity.component.ts

```diff
@@ -73,10 +79,9 @@ export class AssociatedActivityComponent implements OnDestroy, ControlValueAcces
   }
 
   onRemove(): void {
-    if (this.rootActivityId === null) return;
+    if (this.rootActivity === null || this.state !== 'ready') return;
```

Comment:
As it is unexpected, throw an error instead. I was not doing that before but I try to throw errors on unexpected cases so that we can have these unexpected cases  in logs.

Reply:
fixed